pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -1226,6 +1226,40 @@ func (c *Controller) WorkloadInstanceHandler(si *model.WorkloadInstance, event m
 		c.workloadInstancesIndex.Insert(si)
 	}
 
+	// handle WorkloadInstance in ambient by creating a mock pod that models the VM workload.
```

Comment:
I think we need the same logic we use for the rest of Istio? We need ServiceEntries as well, and ServiceEntry selecting Pod, Service selecting WorkloadEntry, etc

Reply:
+1